src/kv1/read.rs

```diff
@@ -560,19 +564,73 @@ where
     Ok(())
 }
 
-fn make_fake_collections(collections: &mut Collections) -> Result<()> {
-    collections.routes = CollectionWithId::new(vec![Route {
-        id: "fake_route".into(),
-        name: "fake_route".into(),
-        direction_type: Some("forward".into()),
-        codes: KeysValues::default(),
-        object_properties: KeysValues::default(),
-        comment_links: CommentLinksT::default(),
-        line_id: "fake_line".into(),
-        geometry_id: None,
-        destination_id: None,
-    }])?;
+fn get_route_origin_destination<'a>(
+    collections: &'a Collections,
+    route_id: &str,
+) -> Option<(&'a StopPoint, &'a StopPoint)> {
+    let stop_times = &collections
+        .vehicle_journeys
+        .values()
+        .filter(|vj| vj.route_id == route_id)
+        .min_by_key(|vj| &vj.id)? // TODO: instead of picking the first trip, find the most frequence origin (or destination) from all the trips
+        .stop_times;
+    let origin = &collections.stop_points[stop_times[0].stop_point_idx];
+    let destination = &collections.stop_points[stop_times[stop_times.len() - 1].stop_point_idx];
+    Some((origin, destination))
+}
 
+fn make_routes(collections: &mut Collections, jopas: &BTreeMap<String, Jopa>) -> Result<()> {
+    for (id, mut group) in &jopas.values().group_by(|j| j.get_route_id()) {
+        let first = group.next().ok_or_else(|| {
```

Comment:
this is false. you need to create routes by direction

Reply:
`get_route_id` returns a `String` based on `LinePlanningNumber` and `Direction`. So each `group` should be unique by direction...